src/components/time-selector/TimeSelector.tsx

```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react'
+import { checkboxArray } from './Checkboxes'
+import { ICheckbox, IDayCard } from './TimeSelectorInterfaces'
+import { TimeSelectorCard } from '../card/TimeSelectorCard'
+
+export const TimeSelector = () => {
+
+    const [cards, setCards] = useState<IDayCard[]>([])
+    const [checkboxes, setCheckboxes] = useState<ICheckbox[]>(checkboxArray)
+
+    const getCards = () => {
+        return ( 
+            cards?.map(card => {
+                const { name } = card
+                return <TimeSelectorCard key={name} name={name}/>
+            })
+        )
+    }
+
+    const markCheckbox = (checkboxValue: boolean, dayName: string, index: number) => {
+        checkboxes[index].checked = checkboxValue
+        setCheckboxes(checkboxes)
+    
+        if(checkboxValue) {
+            setCards([...cards, {name: dayName}])
```

Comment:
The cards needs to be sorted according to the checkboxes. By this setter I think cards would appear randomly.